cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -1015,12 +1079,16 @@ private V1PodSpec createPodSpec(Type resourceType, Location runtimeConfigLocatio
       .getOrDefault(FILE_LOCALIZER_JVM_OPTS, FILE_LOCALIZER_DEFAULT_JVM_OPTS));
 
     // Add volume mounts to the container. Add those from the current pod for mount cdap and hadoop conf.
-    List<V1VolumeMount> volumeMounts = new ArrayList<>(podInfo.getContainerVolumeMounts());
+    // Filter out cconf and hconf volume mounts for jobs because we create config volumes for each pipeline run.
```

Comment:
we shouldn't be looking for specific volume mounts, it violates the abstraction.

What other mounts are here? Do we really need to 'inherit' the volume mounts of the current pod? Isn't everything that needs to be localized specified as a LocalFile?